adapters/beachfront/beachfront.go

```diff
@@ -0,0 +1,541 @@
+//              Video
+// ---------------------------------------------------
+
+type BeachfrontVideoRequest struct {
+	IsPrebid bool                  `json:"isPrebid"`
+	AppId    string                `json:"appId"`
+	Domain   string                `json:"domain"`
+	Id       string                `json:"id"`
+	Imp      []BeachfrontVideoImp  `json:"imp"`
+	Site     BeachfrontSite        `json:"site"`
+	Device   BeachfrontVideoDevice `json:"device"`
+	User     openrtb.User          `json:"user"`
+	Cur      []string              `json:"cur"`
+}
+
+type BeachfrontSite struct {
+	Page string `json:"page"`
+}
+
+type BeachfrontPublisher struct {
+	Id string `json:"id"`
+}
+
+type BeachfrontVideoDevice struct {
+	Ua         string             `json:"ua"`
+	Devicetype int                `json:"deviceType"`
+	Geo        BeachfrontVideoGeo `json:"geo"`
+}
+
+type BeachfrontVideoGeo struct {
+	Ip string `json:"ip"`
+}
+
+type BeachfrontVideoImp struct {
+	Video    BeachfrontSize `json:"video"`
+	Bidfloor float64        `json:"bidfloor"`
+	Id       int            `json:"id"`
+	ImpId    string         `json:"impid"`
+}
+
+// ---------------------------------------------------
+//              Banner
+// ---------------------------------------------------
+type BeachfrontBannerRequest struct {
+	Slots          []BeachfrontSlot `json:"slots"`
+	Domain         string           `json:"domain"`
+	Page           string           `json:"page"`
+	Referrer       string           `json:"referrer"`
+	Search         string           `json:"search"`
+	Secure         int8             `json:"secure"`
+	DeviceOs       string           `json:"deviceOs"`
+	DeviceModel    string           `json:"deviceModel"`
+	IsMobile       int8             `json:"isMobile"`
+	Ua             string           `json:"ua"`
+	Dnt            int8             `json:"dnt"`
+	User           string           `json:"user"`
+	AdapterName    string           `json:"adapterName"`
+	AdapterVersion string           `json:"adapterVersion"`
+	Ip             string           `json:"ip"`
+}
+
+type BeachfrontSlot struct {
+	Slot     string           `json:"slot"`
+	Id       string           `json:"id"`
+	Bidfloor float64          `json:"bidfloor"`
+	Sizes    []BeachfrontSize `json:"sizes"`
+}
+
+type BeachfrontSize struct {
+	W uint64 `json:"w"`
+	H uint64 `json:"h"`
+}
+
+type BeachfrontResponseSlot struct {
+	CrID  string  `json:"crid"`
+	Price float64 `json:"price"`
+	W     uint64  `json:"w"`
+	H     uint64  `json:"h"`
+	Slot  string  `json:"slot"`
+	Adm   string  `json:"adm"`
+}
+
+func (a *BeachfrontAdapter) Name() string {
+	return "beachfront"
+}
+
+func (a *BeachfrontAdapter) SkipNoCookies() bool {
+	return true
+}
+
+func (a *BeachfrontAdapter) MakeRequests(request *openrtb.BidRequest) ([]*adapters.RequestData, []error) {
+	errs := make([]error, 0, len(request.Imp))
+
+	var beachfrontRequests BeachfrontRequests
+	var reqJSON []byte
+
+	a.URI = func() string {
```

Comment:
You have a race condition here. If you have two requests, one for banner and and other for video, come in at the same time, then the two will be overriding the adapter object's URI property. You should make this a local variable so that you don't have this cross talk between `MakeRequests()` calls.